### PR TITLE
fix: Show both sides of the internal transfers in history

### PR DESCRIPTION
**Pull request by obany**
# Description of change

To avoid confusion both side of internal transfers are now shown in history.
The message id has been replaced with more descriptive text about the transaction and associated account.
image

## Links to any relevant issues

Fixes https://github.com/iotaledger/firefly/issues/892

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Tested on windows with external and internal transfers

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/shared/lib/wallet.ts [hunk 1]
```diff
@@ -748,39 +747,20 @@ export const getTransactions = (accounts: WalletAccount[], count = 10): AccountM
...
-            } else {
-                messages[message.id] = Object.assign<
-                    AccountMessage,
-                    Message,
-                    Partial<AccountMessage>
-                >(
-                    {} as AccountMessage,
-                    message,
-                    { account: account.index });
+            messages[account.index + message.id] = {
```
Comment (lucasfernog) [inline packages/shared/lib/wallet.ts:750]: @laumair is this a safe change? i think it is but i just wanna confirm

**Review by lucasfernog — APPROVED** [review]

File: packages/shared/lib/wallet.ts (hunk 1 above)
Comment (laumair) [reply packages/shared/lib/wallet.ts:750]: Yeah

File: packages/shared/locales/en.json [hunk 2]
```diff
@@ -600,7 +600,14 @@
         "passwordStrength3": "Average",
         "passwordStrength4": "Strong",
         "creatingProfile": "Creating profile, please wait...",
-        "accountRemoved": "This account is hidden. Unhide it to perform transfers."
+        "accountRemoved": "This account is hidden. Unhide it to perform transfers.",
+        "receivingTo": "Receiving to",
+        "sendingFrom": "Sending from",
+        "receivedTo": "Received to",
+        "sentFrom": "Sent from",
```
Comment (rajivshah3) [inline packages/shared/locales/en.json:607]:
```suggestion
        "receivingTo": "Receiving to {account}",
        "sendingFrom": "Sending from {account}",
        "receivedTo": "Received to {account}",
        "sentFrom": "Sent from {account}",
```

File: packages/shared/components/ActivityRow.svelte [hunk 3]
```diff
@@ -1,31 +1,87 @@
...
+    <div class="w-8">
+        <Icon
+            boxed
+            classes="text-white dark:text-{payload.data.essence.data.internal ? 'gray-500' : `${color}-${payload.data.essence.data.incoming ? '500' : '600'}`}"
+            boxClasses="bg-{initialsColor ? `${initialsColor}-500` : payload.data.essence.data.internal ? 'gray-500' : `${color}-${payload.data.essence.data.internal ? '500' : '600'}`} dark:bg-gray-900"
+            icon={payload.data.essence.data.internal ? 'transfer' : payload.data.essence.data.incoming ? 'chevron-down' : 'chevron-up'} />
+    </div>
     <div class="flex flex-col ml-3.5 space-y-1.5">
-        <Text type="p" bold smaller>{truncateString(id)}</Text>
+        <Text type="p" bold smaller>{locale(direction)} {accountAlias}</Text>
```
Comment (rajivshah3) [inline packages/shared/components/ActivityRow.svelte:84]:
```suggestion
        <Text type="p" bold smaller>{locale(direction, { values: { account: accountAlias } })}</Text>
```
I'm not a linguistics expert but I believe SOV languages may order this differently, so we should leave the order up to the translator.

**Review by rajivshah3 — CHANGES_REQUESTED** [review]

File: packages/shared/components/ActivityRow.svelte (hunk 3 above)
Comment (obany) [reply packages/shared/components/ActivityRow.svelte:84]: Agreed :)

**Review by rajivshah3 — APPROVED** [review]

**Pull request merged**